homeassistant/components/iaqualink/__init__.py

```diff
@@ -134,23 +139,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         _LOGGER.debug("Got %s switches: %s", len(switches), switches)
         platforms.append(Platform.SWITCH)
 
+    hass.data[DOMAIN]["client"] = aqualink
+
     await hass.config_entries.async_forward_entry_setups(entry, platforms)
 
     async def _async_systems_update(now):
         """Refresh internal state for all systems."""
-        prev = systems[0].online
-
-        try:
-            await systems[0].update()
-        except AqualinkServiceException as svc_exception:
-            if prev is not None:
-                _LOGGER.warning("Failed to refresh iAqualink state: %s", svc_exception)
-        else:
-            cur = systems[0].online
-            if cur is True and prev is not True:
-                _LOGGER.warning("Reconnected to iAqualink")
-
-        async_dispatcher_send(hass, DOMAIN)
+        for system in systems:
+            prev = system.online
+
+            try:
+                await system.update()
+            except AqualinkServiceException as svc_exception:
+                if prev is not None:
+                    _LOGGER.warning(
+                        "Failed to refresh system %s state: %s",
+                        system.serial,
+                        svc_exception,
+                    )
+            else:
+                cur = system.online
+                if cur is True and prev is not True:
+                    _LOGGER.warning("System %s reconnected to iAqualink", system.serial)
+
+            async_dispatcher_send(hass, DOMAIN)
```

Comment:
Side not for the future: I'd recommend switching to use our update coordinator helper instead.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities